File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        int cnt = 0;
        int last = 0;
        for (int ix = 0; ix < nums.size(); ix++) {
            if (cnt == 0)
                last = nums[ix];
            if (nums[ix] == last)
                cnt++;
            else
                cnt--;
        }
        return last;
    }
};

File: 169-majority-element/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        cnt = 0
        curr = 0
        for ix, n in enumerate(nums):
            if cnt == 0:
                curr = n
            if n == curr:
                cnt = cnt + 1
            else:
                cnt = cnt - 1
        
        return curr